File: 942-di-string-match/942-di-string-match.py
class Solution:
    def diStringMatch(self, s: str) -> List[int]:
        n = len(s)
        start = 0
        arr = []
        for i in s:
            if i == 'D':
                arr.append(n) 
                n -= 1
            else:
                arr.append(start)
                start += 1
        arr += [start]
        return arr

File: 942-di-string-match/942-di-string-match.java
class Solution {
    public int[] diStringMatch(String s) {
        int[] arr = new int[s.length()+1];
        int low = 0;
        int high = s.length();
        for (int i = 0; i < s.length();i++){
            if (s.charAt(i) == 'I'){
                arr[i] = low++;
            }else{
                arr[i] = high--;
            }
        }
        arr[s.length()] = high;
        return arr;   
    }
}